src/tcp.c

```diff
@@ -45,14 +44,64 @@ do_connect(struct enftun_tcp* tcp, int mark, struct addrinfo* addr)
 
     enftun_log_debug("TCP: connecting to [%s]:%d\n", ip, port);
 
-    if ((tcp->fd = socket(AF_XPT, SOCK_STREAM, addr->ai_protocol)) < 0)
+    if ((tcp->fd = socket(addr->ai_family, SOCK_STREAM, addr->ai_protocol)) < 0)
```

Comment:
This reverses commit `477038e`. I suggest removing this change and line 32 from this  commit, then remove the other commit entirely.